graph/src/components/store.rs

```diff
@@ -962,6 +962,9 @@ pub trait SubgraphStore: Send + Sync + 'static {
         deployment: &DeploymentLocator,
     ) -> Result<Arc<dyn WritableStore>, StoreError>;
 
+    /// Return a `CursorStore` that is used
+    fn cursor(&self, deployment: &DeploymentLocator) -> Result<Arc<dyn CursorStore>, StoreError>;
```

Comment:
I'd be fine with just returning the cursor from this without going through a separate `CursorStore` unless  you think that `CursorStore` will end up getting passed around a lot and we should therefore not expose all of the subgraph store to those places. But if it's only used in places that already have a `SubgraphStore`, there's no need for the extra indirection.